client/post-editor/editor-slug/index.jsx

```diff
@@ -2,79 +2,84 @@
  * External dependencies
  */
 import ReactDom from 'react-dom';
-import React, { PropTypes } from 'react';
-import PureRenderMixin from 'react-pure-render/mixin';
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
 
 /**
  * Internal Dependencies
  */
-import actions from 'lib/posts/actions';
 import TrackInputChanges from 'components/track-input-changes';
 import FormTextInput from 'components/forms/form-text-input';
 import { recordStat, recordEvent } from 'lib/posts/stats';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostSlug } from 'state/posts/selectors';
+import { editPost } from 'state/posts/actions';
 
-export default React.createClass( {
-	displayName: 'PostEditorSlug',
-
-	mixins: [ PureRenderMixin ],
-
-	propTypes: {
+class PostEditorSlug extends Component {
+	static propTypes = {
 		path: PropTypes.string,
 		slug: PropTypes.string,
 		onEscEnter: PropTypes.func,
 		className: PropTypes.string,
 		isEditable: PropTypes.bool,
-		postType: PropTypes.string,
-		instanceName: PropTypes.string
-	},
-
-	getDefaultProps() {
-		return {
-			onEscEnter: noop,
-			isEditable: true
-		};
-	},
-
-	getInitialState() {
-		return {
-			isSlugFocused: false
-		};
-	},
+		instanceName: PropTypes.string,
+		translate: PropTypes.func
```

Comment:
Finally I can return the favor and call you out on missing `propTypes`! 😆  Missing `siteId` and `postId`